api/image_similarity.py

```diff
@@ -5,19 +5,40 @@
 import numpy as np
 from ownphotos.settings import IMAGE_SIMILARITY_SERVER
 
+def search_similar_embedding(user, emb, result_count=100, threshold=27):
+    if type(user) == int:
+        user_id = user
+    else:
+        user_id = user.id
+
+    image_embedding = np.array(emb, dtype=np.float32)
+
+    post_data = {
+        "user_id":user_id,
+        "image_embedding":image_embedding.tolist(),
+        "n": result_count,
+        "threshold": threshold,
+    }
+    res = requests.post(IMAGE_SIMILARITY_SERVER+'/search/',json=post_data)
+    if res.status_code==200:
+        return res.json()['result']
+    else:
+        logger.error('error retrieving similar embeddings for user {}'.format(user_id))
+        return []
+
 def search_similar_image(user,photo):
     if type(user) == int:
         user_id = user
     else:
         user_id = user.id
     
-    if(photo.encoding == None):
-        photo._im2vec()
-    if(photo.encoding == None):
+    if(photo.clip_embeddings == None):
+        photo._generate_clip_embeddings()
```

Comment:
This line leads to an exception, because the parameter commit is missing.

Reply:
Added a default value to kwarg